mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/store/MantisStorageProvider.java

```diff
@@ -183,4 +186,12 @@ List<MantisWorkerMetadataWritable> getArchivedWorkers(String jobid)
     void deleteExpiredDisableTaskExecutorRequest(DisableTaskExecutorsRequest request) throws IOException;
 
     List<DisableTaskExecutorsRequest> loadAllDisableTaskExecutorsRequests(ClusterID clusterID) throws IOException;
+
+    void storeNewJobArtifact(JobArtifact jobArtifact) throws IOException;
+
+    boolean jobArtifactExists(ArtifactID artifactID);
+
+    List<String> searchJobArtifactProjects(@Nullable String prefix) throws IOException;
```

Comment:
I'm guessing you want to list artifacts by name here?

Reply:
yes. I can better name that.